### DISCUSSION TITLE
Reset/Stop/Motor alarm input

### DISCUSSION THREAD

**Discussion by LucaCortelli** [grblHAL/core#248] [Q&A]
Hello everyone, I'm creating a grblHall PCB with teensy4.1 and I was wondering if in addition to the hard reset you could add a stop pin (softer than the hard reset input) to possibly assign to an alarm coming from the motor drivers, can you advise me something?

**Comment by andrewmarles** [comment]
Yes, you can do this.  The signal is defined in the core, you just need to modify the driver to connect it to an external pin to the "Input_MotorFault" in the crossbar.  You can see in driver.c how the "input_signal_t inputpin[]" are set up for the other input signals.

**Comment by terjeio** [comment]
There are two motor related [input signals defined](https://github.com/grblHAL/core/blob/04ce2c5c8c1129ed98e79b91e4041cd74cd6d08a/system.h#L120-L140), fault and warning. Asserting the fault signal raises alarm 17. The warning signal currently has no default action associated.

In the next build I will add [preprocessor handling](https://github.com/grblHAL/core/blob/master/pin_bits_masks.h) of all input signals in an attempt to ensure consistent naming across drivers, these were missing:
```
ESTOP_PIN
PROBE_DISCONNECT_PIN
STOP_DISABLE_PIN
BLOCK_DELETE_PIN
SINGLE_BLOCK_PIN
MOTOR_FAULT_PIN
MOTOR_WARNING_PIN
LIMITS_OVERRIDE_PIN
```

I'll also add the missing ones to the [crossbar definitions](https://github.com/grblHAL/core/blob/master/crossbar.h).

Code has to be added to drivers to support these inputs - I may add support when requested or accept PRs with the needed code changes.
Note that I will _not_ add all the input options for all drivers as standard as this will require a lot of work that I am not prepared to do...
